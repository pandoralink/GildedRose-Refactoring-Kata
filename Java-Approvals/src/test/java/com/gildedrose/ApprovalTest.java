package com.gildedrose;

import org.approvaltests.Approvals;
import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

class ApprovalTest {

    @Test
    public void updateQuality() {
        CombinationApprovals.verifyAllCombinations(
            this::callUpdateQuality,
            new String[]{"a common item", "Sulfuras, Hand of Ragnaros", "Aged Brie", "Backstage passes to a TAFKAL80ETC concert"},
            new Integer[]{0, -1, 1, 12},
            new Integer[]{0, 2, 49, 50}
        );
    }

    private String callUpdateQuality(String name,int sellIn, int quality) {
        Item[] items = new Item[]{new Item(name, sellIn, quality)};
        GildedRose gildedRose = new GildedRose(items);
        gildedRose.updateQuality();

        return gildedRose.items[0].toString();
    }
}
