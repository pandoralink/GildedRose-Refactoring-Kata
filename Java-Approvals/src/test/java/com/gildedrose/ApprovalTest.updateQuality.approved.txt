[a common item, 0, 0] => a common item, -1, 0 
[a common item, 0, 2] => a common item, -1, 0 
[a common item, 0, 49] => a common item, -1, 47 
[a common item, 0, 50] => a common item, -1, 48 
[a common item, -1, 0] => a common item, -2, 0 
[a common item, -1, 2] => a common item, -2, 0 
[a common item, -1, 49] => a common item, -2, 47 
[a common item, -1, 50] => a common item, -2, 48 
[a common item, 1, 0] => a common item, 0, 0 
[a common item, 1, 2] => a common item, 0, 1 
[a common item, 1, 49] => a common item, 0, 48 
[a common item, 1, 50] => a common item, 0, 49 
[a common item, 12, 0] => a common item, 11, 0 
[a common item, 12, 2] => a common item, 11, 1 
[a common item, 12, 49] => a common item, 11, 48 
[a common item, 12, 50] => a common item, 11, 49 
[Sulfuras, Hand of Ragnaros, 0, 0] => Sulfuras, Hand of Ragnaros, 0, 0 
[Sulfuras, Hand of Ragnaros, 0, 2] => Sulfuras, Hand of Ragnaros, 0, 2 
[Sulfuras, Hand of Ragnaros, 0, 49] => Sulfuras, Hand of Ragnaros, 0, 49 
[Sulfuras, Hand of Ragnaros, 0, 50] => Sulfuras, Hand of Ragnaros, 0, 50 
[Sulfuras, Hand of Ragnaros, -1, 0] => Sulfuras, Hand of Ragnaros, -1, 0 
[Sulfuras, Hand of Ragnaros, -1, 2] => Sulfuras, Hand of Ragnaros, -1, 2 
[Sulfuras, Hand of Ragnaros, -1, 49] => Sulfuras, Hand of Ragnaros, -1, 49 
[Sulfuras, Hand of Ragnaros, -1, 50] => Sulfuras, Hand of Ragnaros, -1, 50 
[Sulfuras, Hand of Ragnaros, 1, 0] => Sulfuras, Hand of Ragnaros, 1, 0 
[Sulfuras, Hand of Ragnaros, 1, 2] => Sulfuras, Hand of Ragnaros, 1, 2 
[Sulfuras, Hand of Ragnaros, 1, 49] => Sulfuras, Hand of Ragnaros, 1, 49 
[Sulfuras, Hand of Ragnaros, 1, 50] => Sulfuras, Hand of Ragnaros, 1, 50 
[Sulfuras, Hand of Ragnaros, 12, 0] => Sulfuras, Hand of Ragnaros, 12, 0 
[Sulfuras, Hand of Ragnaros, 12, 2] => Sulfuras, Hand of Ragnaros, 12, 2 
[Sulfuras, Hand of Ragnaros, 12, 49] => Sulfuras, Hand of Ragnaros, 12, 49 
[Sulfuras, Hand of Ragnaros, 12, 50] => Sulfuras, Hand of Ragnaros, 12, 50 
[Aged Brie, 0, 0] => Aged Brie, -1, 2 
[Aged Brie, 0, 2] => Aged Brie, -1, 4 
[Aged Brie, 0, 49] => Aged Brie, -1, 50 
[Aged Brie, 0, 50] => Aged Brie, -1, 50 
[Aged Brie, -1, 0] => Aged Brie, -2, 2 
[Aged Brie, -1, 2] => Aged Brie, -2, 4 
[Aged Brie, -1, 49] => Aged Brie, -2, 50 
[Aged Brie, -1, 50] => Aged Brie, -2, 50 
[Aged Brie, 1, 0] => Aged Brie, 0, 1 
[Aged Brie, 1, 2] => Aged Brie, 0, 3 
[Aged Brie, 1, 49] => Aged Brie, 0, 50 
[Aged Brie, 1, 50] => Aged Brie, 0, 50 
[Aged Brie, 12, 0] => Aged Brie, 11, 1 
[Aged Brie, 12, 2] => Aged Brie, 11, 3 
[Aged Brie, 12, 49] => Aged Brie, 11, 50 
[Aged Brie, 12, 50] => Aged Brie, 11, 50 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 2] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 49] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 50] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 0] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 2] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 49] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, -1, 50] => Backstage passes to a TAFKAL80ETC concert, -2, 0 
[Backstage passes to a TAFKAL80ETC concert, 1, 0] => Backstage passes to a TAFKAL80ETC concert, 0, 3 
[Backstage passes to a TAFKAL80ETC concert, 1, 2] => Backstage passes to a TAFKAL80ETC concert, 0, 5 
[Backstage passes to a TAFKAL80ETC concert, 1, 49] => Backstage passes to a TAFKAL80ETC concert, 0, 50 
[Backstage passes to a TAFKAL80ETC concert, 1, 50] => Backstage passes to a TAFKAL80ETC concert, 0, 50 
[Backstage passes to a TAFKAL80ETC concert, 12, 0] => Backstage passes to a TAFKAL80ETC concert, 11, 1 
[Backstage passes to a TAFKAL80ETC concert, 12, 2] => Backstage passes to a TAFKAL80ETC concert, 11, 3 
[Backstage passes to a TAFKAL80ETC concert, 12, 49] => Backstage passes to a TAFKAL80ETC concert, 11, 50 
[Backstage passes to a TAFKAL80ETC concert, 12, 50] => Backstage passes to a TAFKAL80ETC concert, 11, 50 
